src/components/MintingButton.tsx

```diff
@@ -0,0 +1,160 @@
+import { useState } from "react"
+import { usePrepareContractWrite, useContractWrite, useWaitForTransaction } from "wagmi"
+import { utils, BigNumber } from "ethers"
+import {
+  Box,
+  Typography,
+  Modal
+} from "@mui/material"
+import { MINT_CONTRACT_ADDRESS, MULTIPLY_GAS_LIMIT } from "config"
+import GenArt721MintABI from "abi/GenArt721Mint.json"
+import TokenView from "components/TokenView"
+import MintingButtonEnabled from "components/MintingButtonEnabled"
+import useWindowSize from "hooks/useWindowSize"
+
+interface Props {
+  projectId: string,
+  priceWei: BigNumber
+  currencySymbol: string,
+  isConnected: boolean,
+  artistCanMint: boolean,
+  anyoneCanMint: boolean,
+  scriptAspectRatio: number
+}
+
+const multiplyBigNumberByFloat = function(x: BigNumber, y: number) {
```

Comment:
Since these are good general functions may be good to pull them out into a utils file somewhere